docs/pipelines.md

```diff
@@ -586,6 +586,20 @@ references the `outputValue` `Result` emitted by the `calculate-sum` `Task`.
 
 For an end-to-end example, see [`Results` in a `PipelineRun`](../examples/v1beta1/pipelineruns/pipelinerun-results.yaml).
 
+A `Pipeline Result` is not emitted if any of the following are true:
+- A `TaskRun` referenced by the `Pipeline Result` failed. The `PipelineRun` will also
+have failed.
+- A `TaskRun` referenced by the `Pipeline Result` was skipped.
+- A `TaskRun` referenced by the `Pipeline Result` didn't emit the referenced `Task Result`. This
+should be considered a bug in the `Task` and [may fail a `TaskRun` in future](https://github.com/tektoncd/pipeline/issues/3497).
+- The `Pipeline Result` uses a variable that doesn't point to a task in the Pipeline. This will
+result in an `InvalidTaskResultReference` validation error during execution of the `Pipeline`.
+- The `Pipeline Result` uses a variable that doesn't point to a result of a task in the Pipeline.
+This will result in an `InvalidTaskResultReference` validation error during execution of the `Pipeline`.
+
+**Note:** Since a `Pipeline Result` can contain references to multiple `Task Results`, if any of those
+`Task Result` references are invalid the entire `Pipeline Result` is not emitted.
```

Comment:
i think as a user i might find it hard to reason about what will happen based on this description, maybe it would be clearer to flip it around a bit, and i think it might be more clear to a user to say something like "task" or "pipeline task" instead of "task run":

PipelineRun execution will fail if a pipeline result requires any of the following:
* A result from a Task that is skipped
* A result from a Task that it declares but when executed it doesn't actually emit it. This should be considered a bug in the `Task` and [may fail a `TaskRun` in future](https://github.com/tektoncd/pipeline/issues/3497).
* A result from a Task that it does not declare
* A result from a Task that does not exit in the Pipeline


I didn't include these ones b/c i'm not sure if they WILL have that same error:
* A `TaskRun` referenced by the `Pipeline Result` failed. The `PipelineRun` will also have failed. <-- since execution will stop in this case, i dont think there would be an error related to the result?

Reply:
OK, I've updated the section to reference "Pipeline Task" throughout instead of "TaskRun".

Inre: your comments on PipelineRun execution behaviour, I think there are at least two discrete questions we're trying to answer in this secion of the doc:

1. What is the final "shape" of a pipeline result if it contains an invalid reference?
2. What happens to pipelinerun execution when pipeline results contain these invalid references?

I think that (1) is answered by the existing PR.  Personally I think that (2) is independent of the results and shouldn't be documented in a section titled `Emitting Results from a Pipeline` since it's orthogonal.  The Pipeline Results don't have any bearing on the execution status of the PipelineRun.  That doesn't mean we shouldn't document it but it does seem to me a little out-of-scope to try and document PipelineRun execution state in a section dedicated to result data.

> PipelineRun execution will fail if a pipeline result requires any of the following:
> 
>* A result from a Task that is skipped

PipelineRun execution doesn't fail when a Task is skipped and a result references it. This is kinda an example of why I'm a tad hesitant to include documentation about pipelinerun execution state in a section dedicated to result data. I think this will come back to bite us if we try and document it here.  One day we'll change the behaviour of PipelineRuns and we'll have to remember that we documented pipelinerun execution in multiple places that aren't `pipelineruns.md` and then comb back through to make sure all those places reflect the changed behaviour.  Personally I'd prefer that we keep the documentation in `pipelines.md` as tightly focused as possible on how Pipeline Results behave.